src/plugins/discover/common/esql_locator.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { DISCOVER_ESQL_LOCATOR } from '@kbn/deeplinks-analytics';
+import { LocatorDefinition, LocatorPublic } from '@kbn/share-plugin/common';
+import { SerializableRecord } from '@kbn/utility-types';
+
+export type DiscoverEsqlLocatorParams = SerializableRecord;
+
+export interface DiscoverEsqlLocatorDependencies {
+  discoverAppLocator: LocatorPublic<SerializableRecord>;
+}
+
+export type DiscoverEsqlLocator = LocatorPublic<DiscoverEsqlLocatorParams>;
+
+export class DiscoverEsqlLocatorDefinition implements LocatorDefinition<DiscoverEsqlLocatorParams> {
+  public readonly id = DISCOVER_ESQL_LOCATOR;
+
+  constructor(protected readonly deps: DiscoverEsqlLocatorDependencies) {}
+
+  public readonly getLocation = async () => {
+    const { discoverAppLocator } = this.deps;
+
+    const params = {
+      query: {
+        esql: `SHOW FUNCTIONS`,
```

Comment:
@stratoula If you could advise as to the best logic for this, we can add it to the locator...?

Reply:
@clintandrewhall this is what we want https://github.com/elastic/kibana/blob/main/src/plugins/discover/public/global_search/search_provider.ts#L52

As Nino proposed we can do the following

1. Retrieve the indices from the api like that https://github.com/elastic/kibana/blob/main/packages/kbn-text-based-editor/src/helpers.ts#L194
2. Check if there is the logs* index. If yes -> 
```
 esql: `from logs* | limit 10`,
```
3. If no --> Get the first index from the list and use this instead
```
 esql: `from ${index} | limit 10`,
```